Document differences between OSP13 and 16 for non-standard networking section (#118)
infrawatch/documentation: doc-Service-Telemetry-Framework/modules/proc_deploying-to-non-standard-network-topologies.adoc

Because a spine-leaf configuration can result in different networks being assigned to different Leaf groupings and those names are typically unique, additional configuration is required in the `parameter_defaults` section of the {Openstack}  environment files.

.Procedure

. Document which networks are available for each of the Leaf roles. For examples of network name definitions, see https://access.redhat.com/documentation/en-us/red_hat_openstack_platform/{vernum}/html-single/spine_leaf_networking/index#creating-a-network-data-file[Creating a network data file] in the _Spine Leaf Networking_ guide. For more information about the creation of the Leaf groupings (roles) and assignment of the networks to those groupings, see https://access.redhat.com/documentation/en-us/red_hat_openstack_platform/{vernum}/html-single/spine_leaf_networking/index#creating-a-roles-data-file[Creating a roles data file] in the _Spine Leaf Networking_ guide.

> . Add the following configuration example to the `ExtraConfig` section for each of the leaf roles. In this example,

Reviewer: Well done on this section, looks like a bit of a nightmare to put together!

+
For the
ifdef::include_when_13[]
`Compute0`
endif::[]
ifdef::include_when_16[]
`ComputeLeaf0`
endif::[]
leaf role, specify extra configuration to perform a hiera lookup to determine which network interface for a particular network to assign to the collectd AMQP host parameter. Perform the same configuration for the QDR listener address parameter.
+
[source,yaml]
ifdef::include_when_13[]
----
Compute0ExtraConfig:
›   tripleo::profile::base::metrics::collectd::amqp_host: "%{hiera('internal_api0')}"
›   tripleo::profile::base::metrics::qdr::listener_addr: "%{hiera('internal_api0')}"
----
endif::[]
ifdef::include_when_16[]
----
ComputeLeaf0ExtraConfig:
›   tripleo::profile::base::metrics::collectd::amqp_host: "%{hiera('internal_api_subnet')}"
›   tripleo::profile::base::metrics::qdr::listener_addr: "%{hiera('internal_api_subnet')}"
----
+
Additional leaf roles typically replace `_subnet` with `_leafN` where `N` represents a unique indentifier for the leaf.
+
----
ComputeLeaf1ExtraConfig:
›   tripleo::profile::base::metrics::collectd::amqp_host: "%{hiera('internal_api_leaf1')}"
›   tripleo::profile::base::metrics::qdr::listener_addr: "%{hiera('internal_api_leaf1')}"
----
endif::[]
+

> On a CephStorage leaf role, an example configuration would be:

Reviewer: ```suggestion
This example configuration is on a CephStorage leaf role:
```
